Admin/PageAdmin.php

```diff
@@ -353,33 +353,48 @@ protected function configureFormFields(FormMapper $formMapper)
         if ($this->hasSubject() && !$this->getSubject()->isInternal()) {
             $formMapper
                 ->with('form_page.group_main_label')
-                    ->add('type', 'sonata_page_type_choice', array('required' => false))
+                    ->add(
+                        'type',
+                        method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix') ?
```

Comment:
@dmeigs this is where "future code" should kick in. Be sure to include a `NEXT_MAJOR` comment so that we know what it is about.